src/object/tcb.c

```diff
@@ -1251,17 +1251,31 @@ exception_t decodeSetMCPriority(cap_t cap, word_t length, word_t *buffer)
 }
 
 #ifdef CONFIG_KERNEL_MCS
-exception_t decodeSetTimeoutEndpoint(cap_t cap, cte_t *slot)
+exception_t decodeSetTimeoutEndpoint(cap_t cap, word_t length, cte_t *slot, word_t *buffer)
 {
-    if (current_extra_caps.excaprefs[0] == NULL) {
-        userError("TCB SetSchedParams: Truncated message.");
+    if (length < 2 || current_extra_caps.excaprefs[0] == NULL) {
```

Comment:
No specific reason from my side, beyond wanting to reuse the existing patterns. It makes verification a lot easier when there is a proof structure that we can reuse.

It's also not immediately obvious to me what it would look like if the parameters are optional. Would it only make sense in this case because they are the only parameters for the function?

Reply:
Only reason to not make them optional is to simplify verification, otherwise it always makes sense, if only for backward compatibility. I can imagine that variable length functions are annoying.

To keep backward compatibility in libsel4 the new function needs a new name, like was done with `seL4_SchedControl_Configure` which was renamed to `seL4_SchedControl_ConfigureFlags`.

Reply:
I'll explore the verification side of things and if the proofs for the variable length functions are straightforward then I'll be happy to go with your suggested change making these parameters optional.

I don't think it's relevant since it wouldn't help with backward compatibility, but an argument could be made that the existing cspace and vspace parameters are very similar to the two new ones and could be made optional as well. However, I imagine that this would lead to `seL4_TCB_SetSpace` having an overly complicated interface and it probably isn't worth exploring.

Reply:
Agreed.